src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/ContestedOrderApprovedLetterService.java

```diff
@@ -53,6 +57,11 @@ public void generateAndStoreContestedOrderApprovedLetter(FinremCaseDetails finre
      * @param authorisationToken the authorisation token for document generation service
      */
     public void generateAndStoreContestedOrderApprovedLetter(FinremCaseDetails finremCaseDetails, String judgeDetails, String authorisationToken) {
+        LocalDate orderApprovedDate = ofNullable(finremCaseDetails.getData().getDraftOrdersWrapper())
+            .map(DraftOrdersWrapper::getExtraReportFieldsInput)
+            .map(ExtraReportFieldsInput::getOrderApprovedDate)
+            .orElse(null);
+        finremCaseDetails.getData().setOrderApprovedDate(orderApprovedDate);
```

Comment:
Could you check if you need the same on `public void generateAndStoreContestedOrderApprovedLetter(CaseDetails caseDetails, String authorisationToken)`? 

And it is better to write a private method to wrap the logic for capturing order approved date .